pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -154,6 +157,21 @@ func (esc *endpointSliceController) onEventInternal(_, ep *v1.EndpointSlice, eve
 	}
 }
 
+func serviceNeedsPush(svc *corev1.Service) bool {
+	if svc.Annotations[annotation.NetworkingExportTo.Name] != "" {
+		namespaces := strings.Split(svc.Annotations[annotation.NetworkingExportTo.Name], ",")
+		exportTo := sets.NewWithLength[visibility.Instance](len(namespaces))
+		for _, ns := range namespaces {
+			ns = strings.TrimSpace(ns)
+			exportTo.Insert(visibility.Instance(ns))
+		}
+		if exportTo.Contains(visibility.None) {
+			return false
```

Comment:
nit: you could just return early on L166 is `ns == visibility.None` instead of building up the map?